src/handle_action.cpp

```diff
@@ -664,6 +664,8 @@ static void smash()
 {
     player &u = g->u;
     map &m = g->m;
+    item *w = &u.weapon;
```

Comment:
Is there a reason why you use a pointer here instead of a reference?
```suggestion
    item &w = u.weapon;
```

Reply:
The weapon being pointed to may change to something else other than your held weapon (may be held gloves or shoes) using a reference, this cannot be done -references cannot be re-assigned. Though thats unimplemented, I'm leaving that open as a possible change in the future.